src/components/CacheFS.brs

```diff
@@ -0,0 +1,63 @@
+' @import /components/rokuComponents/EVPDigest.brs
+function CacheFS() as Object
+  prototype = {}
+
+  prototype._PREFIX = "cachefs:/"
+
+  prototype._digest = Invalid
+  prototype._fileSystem = Invalid
+
+  _constructor = function (m as Object) as Object
+    m._digest = EVPDigest()
+    m._digest.setup("sha1")
+    m._fileSystem = CreateObject("roFileSystem")
+
+    return m
+  end function
+
+  ' Reads data stored in cachefs under the specific key
+  ' @param {String} key
+  ' @returns {Object} Invalid if key is incorrect or is not stored
+  prototype.read = function (key as String) as Object
+    if (key = "") then return Invalid
+
+    filePath = m._PREFIX + m._hash(key)
+    if (NOT m._fileSystem.exists(filePath)) then return Invalid
+
+    content = ReadAsciiFile(filePath)
+
+    return ParseJson(content)
+  end function
+
+  ' Writes data into cachefs under the specific key
+  ' @param {String} key
+  ' @param {Object} data - any value acceptable by native FormatJson function except Invalid
+  ' @returns {Boolean} false if data is not parseable or if storing failed
+  prototype.write = function (key as String, data as Object) as Boolean
+    if (key = "" OR data = Invalid) then return false
```

Comment:
I'm not sure if you can say that data that is `Invalid`, is not parseable.

Reply:
ok, I will add Invalid exception